packages/language-server/src/codeActionProvider.ts

```diff
@@ -0,0 +1,189 @@
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+  CodeActionParams,
+  CodeAction,
+  Diagnostic,
+  DiagnosticSeverity,
+  CodeActionKind,
+  Range,
+} from 'vscode-languageserver'
+import { isNullOrUndefined } from 'util'
+import { getAllRelationNames } from './completion/completions'
+import { convertDocumentTextToTrimmedLineArray } from './MessageHandler'
+
+function insertModelOrEnumAtRange(document: TextDocument): Range {
+  // to insert text into a document create a range where start === end.
+  const start = { line: document.lineCount, character: 0 }
+  return { start: start, end: start }
```

Comment:
Probably necessary, but looks a bit weird that `start` and `end` get the same value.
You can use the short notation here as well
```suggestion
  return { start, end: start }
```